lms/djangoapps/bookmarks/models.py

```diff
@@ -69,3 +85,26 @@ def get_path(block):
         parents_data = parents_data[:2]  # To exclude the unit/vertical block information.
         parents_data.reverse()
         return parents_data
+
+
+class XBlockCache(TimeStampedModel):
+
+    course_key = CourseKeyField(max_length=255, db_index=True)
+    usage_key = LocationKeyField(max_length=255, db_index=True, unique=True)
+
+    display_name = models.CharField(max_length=255, default="")
+    _paths = models.TextField(db_column='paths', null=True, blank=True)
+
+    @property
+    def paths(self):
+        """
+        Parse JSON from the _paths field and return it.
+        """
+        return json.loads(self._paths)
+
+    @path.setter
+    def paths(self, value):
+        """
+        Set data as JSON on the _paths field.
+        """
+        self._path = json.dumps(value)
```

Comment:
I'm up in the air about whether it would be better to create (or find an existing) `JsonBlob` field type, rather than having an `_paths` field, and then a `paths` property. The latter version isn't going to play well with all of the standard django features.